test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1

```diff
@@ -1,49 +1,144 @@
 Describe "Job Cmdlet Tests" -Tag "CI" {
     Context "Simple Jobs" {
         AfterEach {
-            Get-Job | Remove-Job -force
+            Get-Job | Remove-Job -Force
         }
         BeforeEach {
-            $j = start-job -scriptblock { 1 + 1 }
+            $j = Start-Job -ScriptBlock { 1 + 1 } -Name "My Job"
         }
         It "Start-Job produces a job object" {
-            $j.gettype().fullname | should be "System.Management.Automation.PSRemotingJob"
+            $j | Should BeOfType "System.Management.Automation.Job"
+            $j.Name | Should Be "My Job"
         }
         It "Get-Job retrieves a job object" {
-            (Get-Job -id $j.id).gettype().fullname | should be "System.Management.Automation.PSRemotingJob"
+            (Get-Job -Id $j.Id) | Should BeOfType "System.Management.Automation.Job"
+        }
+        It "Get-Job retrieves an array of job objects" {
+            Start-Job -ScriptBlock { 2 * 2 }
+            $jobs = Get-Job
+            $jobs.Count | Should Be 2
+            foreach ($job in $jobs)
+            {
+                $job | Should BeOfType "System.Management.Automation.Job"
+            }
         }
         It "Remove-Job can remove a job" {
-            remove-job $j -force
+            Remove-Job $j -Force
             try {
-                get-job $j -ea Stop
+                Get-Job $j -ErrorAction Stop
                 throw "Execution OK"
             }
             catch {
-                $_.FullyQualifiedErrorId | should be "JobWithSpecifiedNameNotFound,Microsoft.PowerShell.Commands.GetJobCommand"
+                $_.FullyQualifiedErrorId | Should Be "JobWithSpecifiedNameNotFound,Microsoft.PowerShell.Commands.GetJobCommand"
             }
         }
-        It "Receive-Job can retrieve job results" -pending {
-            $waitjob = Wait-Job -Timeout 10 -id $j.id
-            if ( $waitJob ) {
-                $result = receive-job -id $j.id
-                $result.id | Should be 2
-            }
+        It "Receive-Job can retrieve job results" {
+            Wait-Job -Timeout 60 -id $j.id | Should Not BeNullOrEmpty
+            receive-job -id $j.id | Should be 2
+        }
+        It "Start-Job accepts arguments" {
```

Comment:
The test don't use `$j` so we should move it in separate `Context` block without creating a job in `BeforeEach`.